Jellyfin.Networking/Manager/NetworkManager.cs

```diff
@@ -1122,5 +1129,81 @@ private void PrintNetworkInformation(NetworkConfiguration config, bool debug = t
                 _logger.Log(logLevel, "Filter list: {0}", _remoteAddressFilter.Select(s => s.Prefix + "/" + s.PrefixLength));
             }
         }
+
+        /// <summary>
+        /// Initializes the LAN IP address that the server advertises for cast receivers.
+        /// A LAN device like a Chromecast receiver needs an unique LAN address of the Jellyfin server.
+        /// For IPv4, a Private network address (like 192.168.0.0/16 )serves this purpose, and for IPv6
+        /// the Unique Local Address (ULA, fd00::/8) does. An IPv6 global address (2000::/3) also works
+        /// as unique server address but is ranked as less desirable than a local address.
+        /// See e.g. https://en.wikipedia.org/wiki/Private_network
+        ///
+        /// If multiple addresses are available:
+        /// - prefer Unique Local Addresses (ULA) or Site Local Addresses over other (global) addresses.
+        /// - all things equal, prefer IPv4 over IPv6 as the former in practice appears to be more stable.
+        /// </summary>
+        private void InitializeServerLanAddress(NetworkConfiguration config)
+        {
+            lock (_initLock)
+            {
+                int rank = 0;   // ranks most preferable ip address
+                IPAddress ipSrv = IPAddress.None;
+                NetworkInterface[] adapters = NetworkInterface.GetAllNetworkInterfaces();
```

Comment:
We should only consider the ones we're actually binding to

Reply:
Yes I do that by filtering with

https://github.com/Sky-High/jellyfin/blob/653c718c4439f622d1a3dea3e11997f2123ea809/Jellyfin.Networking/Manager/NetworkManager.cs#L1173

This outer/inner loop combination is kind of inefficient but the outerloop is needed to find the adapter matching the interface that are actually bound to, which are in the inner loop.